rocky/katalogus/views/plugin_detail.py

```diff
@@ -88,9 +88,22 @@ def get_oois(self, selected_oois: list[str]) -> dict[str, Any]:
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         context["plugin"] = self.plugin.model_dump()
+        self.check_plugin_type()
         context["plugin_settings"] = self.get_plugin_settings()
         return context
 
+    def check_plugin_type(self):
+        if self.plugin.type != self.task_type:
+            raise Http404("Plugin type does not match url.")
+
+            # it would be nicer if we could redirect.
+            #return redirect(
+            #    reverse(
+            #        f"{self.plugin.type}_detail",
+            #        kwargs={"organization_code": self.organization.code, "plugin_id": self.plugin.id},
+            #    )
+            #)
```

Comment:
Pre-commit complains about this, but I also think we shouldn't add this kind code that doesn't and can't work in comments.

```suggestion
            # It would be nicer if we could redirect, but Django doesn't have an easy way to do that.
            raise Http404("Plugin type does not match url.")
```